src/app/modules/dashboard/components/course-list/course-list.component.ts

```diff
@@ -13,32 +15,39 @@ export class CourseListComponent implements OnInit, OnChanges {
   public coursesStart = 0;
   public coursesCount = 3;
 
+
   @Input() searchValue: string;
 
   constructor(
-    private courseService: CourseService
+    private _courseService: CourseService,
+    private _dataLoaderService: DataLoaderService
   ) { }
 
-  ngOnInit(): void {
-    this.courseService.getList(this.coursesStart, this.coursesCount)
-      .then((courses: ICourse[]) => {
-        this.courses = courses;
-      })
-  }
+  ngOnInit(): void { }
 
   ngOnChanges(changes: SimpleChanges) {
     if (changes.searchValue.currentValue) {
-      this.courseService.getList(this.coursesStart, null, changes.searchValue.currentValue)
-        .then((courses: ICourse[]) => {
-          this.courses = courses;
+      this._courseService.getList(this.coursesStart, null, changes.searchValue.currentValue)
```

Comment:
I suggest to put showDataLoader before this line. Such way our methods will just return data without unexpected side effects. And on this level we can see where we execute loader methods

Reply:
will do